### PR TITLE
8329491: GetThreadListStackTraces function should use JvmtiHandshake

### PR DESCRIPTION
**Pull request by sspitsyn**
The internal JVM TI `JvmtiHandshake` and `JvmtiUnitedHandshakeClosure` classes were introduced in the JDK 22 to unify/simplify the JVM TI functions supporting implementation of the virtual threads. This enhancement is to refactor the JVM TI function `GetThreadListStackTraces` on the base of `JvmtiHandshake` and `JvmtiUnitedHandshakeClosure` classes.

Testing:
 - Ran mach5 tiers 1-6

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329491](https://bugs.openjdk.org/browse/JDK-8329491): GetThreadListStackTraces function should use JvmtiHandshake (**Enhancement** - P4)

### Reviewers
 * [Patricio Chilano Mateo](https://openjdk.org/census#pchilanomate) (@pchilano - **Reviewer**) ⚠️ Review applies to [86cb34e7](https://git.openjdk.org/jdk/pull/18574/files/86cb34e75d35f05a3d9a252b300a77c56e80cdc5)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18574/head:pull/18574` \
`$ git checkout pull/18574`

Update a local copy of the PR: \
`$ git checkout pull/18574` \
`$ git pull https://git.openjdk.org/jdk.git pull/18574/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18574`

View PR using the GUI difftool: \
`$ git pr show -t 18574`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18574.diff">https://git.openjdk.org/jdk/pull/18574.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18574#issuecomment-2031365618)

### CODE REVIEW

File: src/hotspot/share/prims/jvmtiEnvBase.cpp [hunk 1]
```diff
@@ -1983,6 +1983,7 @@ JvmtiHandshake::execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target) {
 
   jvmtiError err = JvmtiEnvBase::get_threadOop_and_JavaThread(tlh.list(), target, &java_thread, &thread_obj);
   if (err != JVMTI_ERROR_NONE) {
+    printf("DBG: JvmtiHandshake::execute: err: %d\n", (int)err); fflush(0);
```
Comment (pchilano) [inline src/hotspot/share/prims/jvmtiEnvBase.cpp:1986]: Any reason why not use UL instead with jvmti tag?

**Review by pchilano — APPROVED** [review]
Looks good to me.

**Comment by sspitsyn** [comment]
Patricio, thank you for prompt review!
Your question was removed with my push but it was a nice catch.
 The `printf` was a left over that was temporarily used for debugging. Removed now.

File: src/hotspot/share/prims/jvmtiEnvBase.cpp (hunk 1 above)
Comment (sspitsyn) [reply src/hotspot/share/prims/jvmtiEnvBase.cpp]: Nice catch. The `printf` was temporarily used for debugging. Removed now.

**Review by lmesnik — CHANGES_REQUESTED** [review]

File: src/hotspot/share/prims/jvmtiEnvBase.cpp [hunk 2]
```diff
@@ -2053,17 +2053,28 @@ VM_GetThreadListStackTraces::doit() {
...
     ResourceMark rm;
     _collector.fill_frames(_jthread, jt, thread_oop);
     _collector.allocate_and_fill_stacks(1);
+    set_result(_collector.result());
   }
 }
 
+void
+GetSingleStackTraceClosure::do_thread(Thread *target) {
+  doit();
```
[the remark this answers is not in the document]
Comment (sspitsyn) [reply src/hotspot/share/prims/jvmtiEnvBase.cpp:2070]: Thank you for suggestion. Added asserts.

**Review by lmesnik — APPROVED** [review]

**Comment by sspitsyn** [comment]
Leonid, thank you for review!

**Pull request closed without merging**